src/cli/hook-utils/get-check-update-data.js

```diff
@@ -0,0 +1,334 @@
+    }
+  } catch (err) {
+    throw new Error(err);
+  }
+  return fileContents;
+}
+
+/**
+ * Gathers all related dependency files for the CLI project (package.json).
+ * @param cwd the current working directory of the CLI project
+ */
+async function gatherDependencyFiles(cwd) {
+  const { jsonDepFiles, inaccessibleFiles } = await getJSONFiles(cwd);
+  const dependencyFiles = jsonDepFiles;
+  return { dependencyFiles, inaccessibleFiles };
+}
+
+/**
+ * Gets the needed files that contain dependency info (package.json).
+ * @param cwd the current working directory of the CLI project
+ */
+async function getJSONFiles(cwd) {
+  const jsonFiles = ['package.json'];
+  const jsonDepFiles = [];
+  const inaccessibleFiles = [];
+
+  for (const fileName of jsonFiles) {
+    try {
+      const jsonFile = await getJSON(`${cwd}/${fileName}`);
+      const jsonIsParsable =
+        jsonFile &&
+        typeof jsonFile === 'object' &&
+        !Array.isArray(jsonFile) &&
+        jsonFile.dependencies;
+
+      if (jsonIsParsable) {
+        jsonDepFiles.push(fileName);
+      }
+    } catch (err) {
+      inaccessibleFiles.push({ name: fileName, error: err });
+    }
+  }
+
+  return { jsonDepFiles, inaccessibleFiles };
+}
+
+/**
+ * Pulls dependencies from a given file and JSON structure.
+ * @param json JSON information that includes dependencies
+ * @param fileName name of the file that the dependency list is coming from
+ */
+dependencyExports.extractDependencies = async (json, fileName) => {
+  // Determine if the JSON passed is an object
+  const jsonIsParsable =
+    json !== null && typeof json === 'object' && !Array.isArray(json);
+
+  if (jsonIsParsable) {
+    let boltCurrentVersion = '';
+    let denoCurrentVersion = '';
+    if (json['dependencies']['@slack/bolt']) {
+      const boltCurrentVersionOutput = JSON.parse(
+        await dependencyExports.getBoltCurrentVersion()
+      );
+
+      if (boltCurrentVersionOutput !== '') {
+        boltCurrentVersion =
+          boltCurrentVersionOutput['dependencies']['@slack/bolt']['version'];
+      }
+    }
+    if (json['dependencies']['@slack/deno-slack-sdk']) {
+      const denoCurrentVersionOutput = JSON.parse(
+        await dependencyExports.getDenoCurrentVersion()
+      );
+
+      if (denoCurrentVersionOutput !== '') {
+        denoCurrentVersion =
+          denoCurrentVersionOutput['dependencies']['@slack/deno-slack-sdk'][
+            'version'
+          ];
+      }
+    }
+
+    return [
+      [
+        '@slack/bolt',
+        {
+          version: boltCurrentVersion,
+        },
+      ],
+      [
+        '@slack/deno-slack-sdk',
+        {
+          version: denoCurrentVersion,
+        },
+      ],
+    ];
+  }
+
+  return [];
+};
+
+/**
+ * Gets the latest module version.
+ * @param moduleName the module that the latest version is being queried for
+ */
+async function fetchLatestModuleVersion(moduleName) {
+  let command = "";
```

Comment:
this was originally an API call to NPM but was updated to pull from `npm info` - would love to hear if anyone has any preferences for this!